src/python/WMCore/WMSpec/Steps/Executors/CMSSW.py

```diff
@@ -241,24 +244,25 @@ def execute(self, emulator=None):
 
         os.environ.update(envOverride)
 
-        returncode = subprocess.call(args, stdout=stdoutHandle, stderr=stderrHandle)
+        returnCode = subprocess.call(args, stdout=stdoutHandle, stderr=stderrHandle)
+        returnMessage = None
 
-        if returncode != 0:
+        if returnCode != 0:
             argsDump = {'arguments': args}
             msg = "Error running cmsRun\n%s\n" % argsDump
             try:
                 self.report.parse(jobReportXML, stepName=self.stepName)
-                returncode = self.report.getStepExitCode(stepName=self.stepName)
-                msg += "CMSSW Return code: %s\n" % returncode
+                (returnCode, returnMessage) = self.report.getStepExitCodeAndMessage(stepName=self.stepName)
+                msg += "CMSSW Return code: %s\n" % returnCode
             except Exception as ex:
                 # If report parsing fails, report linux exit code
-                msg += "Linux Return code: %s\n" % returncode
+                msg += "Linux Return code: %s\n" % returnCode
             finally:
                 logging.critical(msg)
```

Comment:
How about adding a `logging.critical("Error message: %s", returnMessage)` in the next line?
Sorry for not suggesting it in the previous review.

Reply:
Done